### PR TITLE
[FEATURE] Add "Reduce LR On Plateau" scheduler 

### PR DESCRIPTION
**Pull request by haven-jeon**
## Description ##

The LR scheduler that reduce learning rate when a metric has stopped improving. #887
  
Modified code from https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.ReduceLROnPlateau

### Essentials ###
- [v] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [v] Changes are complete (i.e. I finished coding on this PR)
- [v] All changes have test coverage
- [v] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by mli** [comment]
Job PR-897/1 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-897/1/index.html

File: src/gluonnlp/lr_scheduler/reduce_lr_on_plateau.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        if threshold_mode not in {'rel', 'abs'}:
+            raise ValueError('threshold mode ' + threshold_mode +
+                             ' is unknown!')
+
+        if mode == 'min':
+            self.mode_worse = np.Inf
+        else:  # mode == 'max':
+            self.mode_worse = -np.Inf
+
+        self.is_better = partial(self._cmp, mode, threshold_mode, threshold)
```
Comment (szhengac) [inline src/gluonnlp/lr_scheduler/reduce_lr_on_plateau.py]: Why do you want to use partial? `mode, threshold_mode, threshold` are simply class variables, so you can simply use them by adding `self.` to the prefix.

File: src/gluonnlp/lr_scheduler/reduce_lr_on_plateau.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+            self.trainer.set_learning_rate(new_lr)
+            if self.verbose:
+                print('Epoch {:5d}: reducing learning rate'
+                      ' {} to {}.'.format(epoch, old_lr, new_lr))
+
+    @property
+    def in_cooldown(self):
+        return self.cooldown_counter > 0
+
+    def _cmp(self, mode, threshold_mode, threshold, a, best):
```
Comment (szhengac) [inline src/gluonnlp/lr_scheduler/reduce_lr_on_plateau.py]: The current design does not look very scalable. That means it requires hard coding if we would like add some changes/schedules.

File: src/gluonnlp/lr_scheduler/reduce_lr_on_plateau.py (hunk 1 above)
Comment (haven-jeon) [reply src/gluonnlp/lr_scheduler/reduce_lr_on_plateau.py]: Right.  I will fix.

File: src/gluonnlp/lr_scheduler/reduce_lr_on_plateau.py (hunk 2 above)
Comment (haven-jeon) [reply src/gluonnlp/lr_scheduler/reduce_lr_on_plateau.py]: Thanks for the comments.
I can consider more flexible class desine.

**Comment by szha** [comment]
@haven-jeon gentle ping
